crates/driver/src/boundary/settlement.rs

```diff
@@ -167,41 +165,39 @@ impl Settlement {
     }
 }
 
-fn into_boundary_order(order: &competition::auction::Order) -> Order {
+fn to_boundary_order(order: &competition::Order) -> Order {
     Order {
         data: OrderData {
             sell_token: order.sell.token.into(),
             buy_token: order.buy.token.into(),
             sell_amount: order.sell.amount,
             buy_amount: order.buy.amount,
-            fee_amount: order.fee.user.into(),
+            // TODO Not sure if this was an oversight by Nick or if this needs to change
+            fee_amount: Default::default(),
```

Comment:
My understanding is that the `competition::Order` domain object needs to contain both the `fee_amount` that is signed by the user and the `full_fee_amount` which is used for the objective value computation (and gets used by the DTO between `driver` and `solver/searcher` here).